kitsune/users/api.py

```diff
@@ -54,26 +53,48 @@ def usernames(request):
     if not request.user.is_authenticated:
         return []
 
-    profiles = Profile.objects.filter(Q(name__istartswith=pre)).values_list("user_id", flat=True)
+    profile_ids = list(
+        Profile.objects.filter(Q(name__istartswith=pre)).values_list("user_id", flat=True)[:10]
+    )
     users = (
-        User.objects.filter(Q(username__istartswith=pre) | Q(id__in=profiles))
-        .extra(select={"length": "Length(username)"})
-        .order_by("length")
+        User.objects.filter(Q(username__istartswith=pre) | Q(id__in=profile_ids))
+        .filter(is_active=True)
         .select_related("profile")
```

Comment:
You can tweak this a bit more by having:
```python
profile_names = list(Profile.objects.filter(Q(name__icontains=pre).values_list('name')[:10])
users = (
    User.objects.filter(Q(username__icontains=pre) | Q(profile__name__in=profile_names))
    .filter(is_active=True)
    .select_related('profile')
)
```

Reply:
This reduces the query 30% more for me

Reply:
Another interesting approach would be to use `Subquery` and fetch everything in one go. Unfortunately mysql does not support `limit` in subqueries

Reply:
> Another interesting approach would be to use Subquery and fetch everything in one go. Unfortunately mysql does not support limit in subqueries

Indeed that was my first approach. Performance is also :hankey: in subqueries in mysql - the django docs actually suggest this approach: https://docs.djangoproject.com/en/2.2/ref/models/querysets/#nested-queries-performance